backend/controllers/prisonerProfile/sentenceAndReleaseViewModels/courtCasesViewModel.js

```diff
@@ -0,0 +1,101 @@
+const moment = require('moment')
+
+const { formatCurrency, readableDateFormat } = require('../../../utils')
+
+const onlyValidValues = value => Boolean(value)
+
+const mergeMostRecentLicenceTerm = sentences =>
+  sentences.reduce((result, current) => {
+    if (current.sentenceTermCode === 'IMP' && !result) return current
+    if (current.sentenceTermCode === 'LIC' && !result.licence) {
+      return {
+        licence: {
+          years: current.years,
+          months: current.months,
+          days: current.days,
+        },
+        ...result,
+      }
+    }
+
+    return result
+  }, null)
+
+const groupSentencesBySequence = sentences =>
+  sentences.reduce((result, current) => {
+    const key = current.sentenceSequence
+    const existing = result.find(sentence => sentence.key === key)
+
+    if (existing) {
+      return [{ ...existing, items: [...existing.items, current] }, ...result.filter(entry => entry.key !== key)]
```

Comment:
A JS lesson for me here... when destructuring `existing` and then adding the `items` explicitly, does this overwrite what is brought in by `existing`? _Goes off to investigate...!_

Reply:
... yes it does 😄

Reply:
Ha, you're too fast.

Reply:
Essentially what's happening here:
`items: [...existing.items, current] }`
Is were are creating a new array with the existing items plus one more item (current)